test/prototype-pollution.js

```diff
@@ -4,6 +4,23 @@
 const test = require('tape');
 const { parseArgs } = require('../index.js');
 
+// These tests are not synced upstream with node, in case of possible side-effects.
+// See index.js for tests shared with upstream.
+
+function setObjectPrototype(prop, value) {
+  const oldValue = Object.prototype[prop];
+  Object.prototype[prop] = value;
+  return oldValue;
+}
+
+function restoreObjectPrototype(prop, oldValue) {
+  if (oldValue == null) {
+    delete Object.prototype[prop];
+  } else {
+    Object.prototype[prop] = oldValue;
```

Comment:
this doesn't properly restore the original property descriptor; instead of using a Get and an assignment, this should use an `Object.getOwnPropertyDescriptor` and an `Object.defineProperty`.

(which reminds me, i need to make a package for this pattern)

Reply:
Oh, I learnt about those today, I see what you mean!